### PR TITLE
[yle_areena] Extended to proceed protected content.

### LINKED ISSUE #5734: Yle areena: some media are not supported
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Europe

### Example URLs

https://areena.yle.fi/1-4266128

### Provide a description that is worded well enough to be understood

Extractor fails if a video without episode_number defined in JSON metadate is downloaded.

At the same time, the extractor does not allow downloading videos marked as restricted.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://areena.yle.fi/1-4266128']
[debug] Encodings: locale cp1250, fs utf-8, pref cp1250, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b644025b] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 0e96b408b
[debug] Python 3.11.0 (CPython AMD64 64bit) - Windows-10-10.0.18363-SP0 (OpenSSL 1.1.1q  5 Jul 2022)
[debug] exe versions: ffmpeg 4.4 (setts), ffprobe 4.4
[debug] Optional libraries: certifi-2022.09.24, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1731 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [YleAreena] Extracting URL: https://areena.yle.fi/1-4266128
[YleAreena] 1-4266128: Downloading webpage
[YleAreena] 1-4266128: Downloading JSON metadata
ERROR: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
Traceback (most recent call last):
  File "C:\yt-dlp\yt_dlp\YoutubeDL.py", line 1495, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\yt-dlp\yt_dlp\YoutubeDL.py", line 1571, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "C:\yt-dlp\yt_dlp\extractor\common.py", line 678, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\yt-dlp\yt_dlp\extractor\yle_areena.py", line 65, in _real_extract
    or int(season_number)),
       ^^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

### PR DESCRIPTION
**Pull request by docbender**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

ADD DESCRIPTION HERE

Some site videos does not provide inside JSON metadata episode and season numbers (Variables episode_number and season_number are null). This will cause the extractor to crash due strict number conversion.
It's also not possible to download restricted videos. These videos require the http header to be properly filled and a referrer is required by Kaltura API. 
This PR addresses both of these issues. New test for extractor is included.

Fixes #5734

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/yle_areena.py [hunk 1]
```diff
@@ -1,40 +1,82 @@
...
         video_id = self._match_id(url)
         info = self._search_json_ld(self._download_webpage(url, video_id), video_id, default={})
+        # Generate headers
+        real_headers = self.generate_api_headers()
+        real_headers.update({'content-type': 'application/json'})
+
         video_data = self._download_json(
             f'https://player.api.yle.fi/v1/preview/{video_id}.json?app_id=player_static_prod&app_key=8930d72170e48303cf5f3867780d549b',
-            video_id)
+            video_id, fatal=True, headers=real_headers)
```
Comment (pukkandan) [inline yt_dlp/extractor/yle_areena.py:79]: why not just `headers={...}`?

File: yt_dlp/extractor/yle_areena.py [hunk 2]
```diff
@@ -52,20 +94,33 @@ def _real_extract(self, url):
...
+        release_date = '%s%s%s' % (start_time[0:4], start_time[5:7], start_time[8:10])
+        # Service URL
+        kaltura_url = 'kaltura:1955031:%s' % traverse_obj(video_data, ('data', 'ongoing_ondemand', 'kaltura', 'id'))
+        # Include video source URL for Kaltura internal use
+        smuggle_data = smuggle_url(kaltura_url, {'source_url': url})
+
         return {
             '_type': 'url_transparent',
-            'url': 'kaltura:1955031:%s' % traverse_obj(video_data, ('data', 'ongoing_ondemand', 'kaltura', 'id')),
+            'url': smuggle_data,
```
Comment (pukkandan) [inline yt_dlp/extractor/yle_areena.py:109]:
```suggestion

        return {
            '_type': 'url_transparent',
            'url': smuggle_url(
                f'kaltura:1955031:{video_data["data"]["ongoing_ondemand"]["kaltura"]["id"]}',
                {'source_url': url}),
```

File: yt_dlp/extractor/yle_areena.py [hunk 3]
```diff
@@ -52,20 +94,33 @@ def _real_extract(self, url):
...
-                              or int(season_number)),
+                              or int_or_none(season_number)),
             'episode_number': (traverse_obj(video_data, ('data', 'ongoing_ondemand', 'episode_number'), expected_type=int_or_none)
-                               or int(episode_number)),
+                               or int_or_none(episode_number)),
             'thumbnails': traverse_obj(info, ('thumbnails', ..., {'url': 'url'})),
             'age_limit': traverse_obj(video_data, ('data', 'ongoing_ondemand', 'content_rating', 'age_restriction'), expected_type=int_or_none),
             'subtitles': subtitles,
+            'manifest_url': manifest_url,
+            'webpage_url': url,
```
Comment (pukkandan) [inline yt_dlp/extractor/yle_areena.py:124]: webpage_url is auto-generated. manifest_url makes no sense for `_type=url_transparent`
```suggestion
```

File: yt_dlp/extractor/yle_areena.py [hunk 4]
```diff
@@ -52,20 +94,33 @@ def _real_extract(self, url):
                     'name': sub.get('kind'),
                 })
 
+        # Video manifest URL
+        manifest_url = traverse_obj(video_data, ('data', 'ongoing_ondemand', 'manifest_url'))
+        # Extract video release date
+        start_time = traverse_obj(video_data, ('data', 'ongoing_ondemand', 'start_time'), expected_type=str)
+        release_date = '%s%s%s' % (start_time[0:4], start_time[5:7], start_time[8:10])
```
Comment (pukkandan) [inline yt_dlp/extractor/yle_areena.py:101]: Use `utils.unified_strdate`

File: yt_dlp/extractor/yle_areena.py [hunk 5]
```diff
@@ -52,20 +94,33 @@ def _real_extract(self, url):
...
+        # Service URL
+        kaltura_url = 'kaltura:1955031:%s' % traverse_obj(video_data, ('data', 'ongoing_ondemand', 'kaltura', 'id'))
+        # Include video source URL for Kaltura internal use
+        smuggle_data = smuggle_url(kaltura_url, {'source_url': url})
```
Comment (pukkandan) [inline yt_dlp/extractor/yle_areena.py:105]: Unnecessary single use variables

File: yt_dlp/extractor/yle_areena.py (hunk 1 above)
Comment (pukkandan) [inline yt_dlp/extractor/yle_areena.py:79]:
```suggestion
            video_id, headers=real_headers)
```
it is the default

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/yle_areena.py (hunk 1 above)
Comment (docbender) [reply yt_dlp/extractor/yle_areena.py]: You're right. A relic from the testing.

Comment (pukkandan) [reply yt_dlp/extractor/yle_areena.py]: Seems you misunderstood
```py
video_id, headers={
    'Origin': 'https://areena.yle.fi/',
    'Referer': 'https://areena.yle.fi/',
    'Content-type': 'application/json',
}
```

**Pull request merged**